src/main/java/org/zalando/nakadi/config/ValidatorConfig.java

```diff
@@ -65,6 +66,7 @@ public SchemaEvolutionService schemaEvolutionService() throws IOException {
         final Schema metaSchema = SchemaLoader.load(metaSchemaJson);
 
         final List<SchemaEvolutionConstraint> schemaEvolutionConstraints = Lists.newArrayList(
+                new CategoryChangeConstraint(),
```

Comment:
I think that the name of the class might be misleading. Every check done here is event type related: partition keys, enrichment strategies, partition strategy, compatibility mode and now categories.